packages/crypto/src/key_pair_ed25519.ts

```diff
@@ -0,0 +1,59 @@
+import { baseEncode, baseDecode } from 'borsh';
+import nacl from 'tweetnacl';
+
+import { KeyType } from './constants';
+import { KeyPairBase, Signature} from './key_pair_base';
+import { PublicKey } from './public_key';
+
+/**
+ * This class provides key pair functionality for Ed25519 curve:
+ * generating key pairs, encoding key pairs, signing and verifying.
+ */
+export class KeyPairEd25519 extends KeyPairBase {
```

Comment:
By extending `KeyPairBase` rather than `KeyPair`, this becomes a breaking change by no longer including the static `fromString` method now on the abstract `KeyPair` class (which is now a sibling to `KeyPairEd25519`). Use of this now-deprecated method in practice would be circuitous and confusing, but its removal is still a breaking change.

This change isn't strictly necessary but is the only way to break the circular dependency across `PublicKey`, `KeyPair`, and `KeyPairEd25519`. This restructuring enables a more substantial decoupling later on, which involves moving `PublicKey` to `@near-js/types` and consolidating all `ed25519` library usage in this module.